tensorboard/plugins/core/core_plugin.py

```diff
@@ -93,15 +96,39 @@ def _serve_asset(self, path, gzipped_asset_bytes, request):
     return http_util.Respond(
         request, gzipped_asset_bytes, mimetype, content_encoding='gzip')
 
+  @wrappers.Request.application
+  def _serve_environment(self, request):
+    """Serve a JSON object containing some base properties used by the frontend.
+
+    * data_location is eitgher a path to a directory or an address to a
+      database (depending on which mode TensorBoard is running in).
+    * window_title is the title of the TensorBoard web page.
+    """
+    return http_util.Respond(
+        request,
+        {
+            'data_location': self._logdir or self._db_uri,
+            'window_title': self._window_title,
+        },
+        'application/json')
+
   @wrappers.Request.application
   def _serve_logdir(self, request):
-    """Respond with a JSON object containing this TensorBoard's logdir."""
+    """Respond with a JSON object containing this TensorBoard's logdir.
+
+    TODO(chihuahua): Remove this method once the frontend instead uses the
+    /environment route (and no deps throughout Google use the /logdir route).
```

Comment:
nit: let's call it `/data/environment` (and `/data/logdir`) here and in the comment on `_serve_window_properties` just for maximal clarity

Reply:
Done.